src/Microsoft.ML.Transforms/EntryPoints/TextAnalytics.cs

```diff
@@ -158,5 +158,18 @@ public static CommonOutputs.TransformOutput WordEmbeddings(IHostEnvironment env,
                 OutputData = view
             };
         }
+
+        internal sealed class ReadonlyMemoryComparer : IEqualityComparer<ReadOnlyMemory<char>>
+        {
+            public bool Equals(ReadOnlyMemory<char> x, ReadOnlyMemory<char> y)
+            {
+                return x.Span.SequenceEqual(y.Span);
+            }
+
+            public int GetHashCode(ReadOnlyMemory<char> obj)
+            {
+                return obj.ToString().GetHashCode();
```

Comment:
I assume the .ToString() is needed? 

ReadOnlyMemory has a .GetHashCode():
https://docs.microsoft.com/en-us/dotnet/api/system.readonlymemory-1.gethashcode?view=netcore-2.1

Reply:
The ReadOnlyMemory<T> implementation calls GetHashCode on a T[], and I'm not sure what that does. This is just to make sure that two different instances that represent the same string get hashed to the same value.

---
In reply to: [234101834](https://github.com/dotnet/machinelearning/pull/1641#discussion_r234101834) [](ancestors = 234101834)